// ngx-bootstrap-slider/bundles/ngx-bootstrap-slider.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('ngx-bootstrap-slider', ['exports', '@angular/core'], factory) :
    (global = global || self, factory(global['ngx-bootstrap-slider'] = {}, global.ng.core));
}(this, (function (exports, core) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    /* global Reflect, Promise */

    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };

    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    function __rest(s, e) {
        var t = {};
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
            t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }

    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [op[0] & 2, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    function __exportStar(m, exports) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }

    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m) return m.call(o);
        if (o && typeof o.length === "number") return {
            next: function () {
                if (o && i >= o.length) o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }

    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m) return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
        }
        catch (error) { e = { error: error }; }
        finally {
            try {
                if (r && !r.done && (m = i["return"])) m.call(i);
            }
            finally { if (e) throw e.error; }
        }
        return ar;
    }

    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }

    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    };

    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }

    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length) resume(q[0][0], q[0][1]); }
    }

    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }

    function __asyncValues(o) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function(v) { resolve({ value: v, done: d }); }, reject); }
    }

    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
        return cooked;
    };

    function __importStar(mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
        result.default = mod;
        return result;
    }

    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }

    function __classPrivateFieldGet(receiver, privateMap) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to get private field on non-instance");
        }
        return privateMap.get(receiver);
    }

    function __classPrivateFieldSet(receiver, privateMap, value) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to set private field on non-instance");
        }
        privateMap.set(receiver, value);
        return value;
    }

    var NgxBootstrapSliderService = /** @class */ (function () {
        function NgxBootstrapSliderService() {
        }
        NgxBootstrapSliderService.ɵprov = core["ɵɵdefineInjectable"]({ factory: function NgxBootstrapSliderService_Factory() { return new NgxBootstrapSliderService(); }, token: NgxBootstrapSliderService, providedIn: "root" });
        NgxBootstrapSliderService = __decorate([
            core.Injectable({
                providedIn: 'root'
            }),
            __metadata("design:paramtypes", [])
        ], NgxBootstrapSliderService);
        return NgxBootstrapSliderService;
    }());

    var NgxBootstrapSliderComponent = /** @class */ (function () {
        function NgxBootstrapSliderComponent() {
        }
        NgxBootstrapSliderComponent.prototype.ngOnInit = function () {
        };
        NgxBootstrapSliderComponent = __decorate([
            core.Component({
                selector: 'mv-ngx-bootstrap-slider',
                template: "\n    <p>\n      ngx-bootstrap-slider works!\n    </p>\n  "
            }),
            __metadata("design:paramtypes", [])
        ], NgxBootstrapSliderComponent);
        return NgxBootstrapSliderComponent;
    }());

    var SliderComponent = /** @class */ (function () {
        function SliderComponent() {
            this.initialOptions = {};
            this.valueChange = new core.EventEmitter();
            // The following events are emitted
            this.slide = new core.EventEmitter();
            this.slideStart = new core.EventEmitter();
            this.slideStop = new core.EventEmitter();
            this.change = new core.EventEmitter();
        }
        Object.defineProperty(SliderComponent.prototype, "style", {
            // This property is not applied to the underlying slider element
            set: function (value) {
                if (this.slider) {
                    this.slider.getElement().setAttribute('style', value);
                }
                else {
                    this.initialStyle = value;
                }
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "value", {
            // Handle the value (two-way)
            set: function (value) {
                if (this.slider) {
                    this.slider.setValue(value);
                }
                else {
                    this.initialOptions.value = value;
                }
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "min", {
            // The following properties are applied to the underlying slider element
            set: function (value) {
                this.changeAttribute('min', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "max", {
            set: function (value) {
                this.changeAttribute('max', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "step", {
            set: function (value) {
                this.changeAttribute('step', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "precision", {
            set: function (value) {
                this.changeAttribute('precision', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "orientation", {
            set: function (value) {
                this.changeAttribute('orientation', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "range", {
            set: function (value) {
                this.changeAttribute('range', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "selection", {
            set: function (value) {
                this.changeAttribute('selection', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "tooltip", {
            set: function (value) {
                this.changeAttribute('tooltip', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "tooltipSplit", {
            set: function (value) {
                this.changeAttribute('tooltip_split', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "tooltipPosition", {
            set: function (value) {
                this.changeAttribute('tooltip_position', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "handle", {
            set: function (value) {
                this.changeAttribute('handle', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "reversed", {
            set: function (value) {
                this.changeAttribute('reversed', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "rtl", {
            set: function (value) {
                this.changeAttribute('rtl', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "enabled", {
            set: function (value) {
                this.changeAttribute('enabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "naturalArrowKeys", {
            set: function (value) {
                this.changeAttribute('natural_arrow_keys', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "ticks", {
            set: function (value) {
                this.changeAttribute('ticks', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "ticksPositions", {
            set: function (value) {
                this.changeAttribute('ticks_positions', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "ticksLabels", {
            set: function (value) {
                this.changeAttribute('ticks_labels', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "ticksSnapBounds", {
            set: function (value) {
                this.changeAttribute('ticks_snap_bounds', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "ticksTooltip", {
            set: function (value) {
                this.changeAttribute('ticks_tooltip', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "scale", {
            set: function (value) {
                this.changeAttribute('scale', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "labelledBy", {
            set: function (value) {
                this.changeAttribute('labelledby', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "rangeHighlights", {
            set: function (value) {
                this.changeAttribute('rangeHighlights', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "formatter", {
            set: function (value) {
                this.changeAttribute('formatter', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(SliderComponent.prototype, "lockToTicks", {
            set: function (value) {
                this.changeAttribute('lock_to_ticks', value);
            },
            enumerable: true,
            configurable: true
        });
        SliderComponent.prototype.addChangeListeners = function () {
            var _this = this;
            this.slider.on('slide', function (value) {
                _this.slide.emit(value);
            });
            this.slider.on('slideStart', function (value) {
                _this.slideStart.emit(value);
            });
            this.slider.on('slideStop', function (value) {
                _this.slideStop.emit(value);
            });
            this.slider.on('change', function (event) {
                _this.change.emit(event);
                _this.valueChange.emit(event.newValue);
            });
        };
        SliderComponent.prototype.prepareSlider = function () {
            // We need to add the change listeners again after each refresh
            this.addChangeListeners();
            // Add the styling to the element
            this.slider.getElement().setAttribute('style', this.initialStyle);
        };
        SliderComponent.prototype.changeAttribute = function (name, value) {
            if (this.slider) {
                this.slider.setAttribute(name, value);
                this.slider.refresh();
                this.prepareSlider();
            }
            else {
                this.initialOptions[name] = value;
            }
        };
        SliderComponent.prototype.ngOnInit = function () {
            // Don't set the value over the initial options, because it will alway
            // be reset. This option seems to be a little buggy.
            var value = undefined;
            if (this.initialOptions.value) {
                value = this.initialOptions.value;
                delete this.initialOptions['value'];
            }
            this.slider = new Slider(this.sliderElement.nativeElement, this.initialOptions);
            if (value) {
                this.slider.setValue(value);
            }
            this.prepareSlider();
        };
        __decorate([
            core.ViewChild('slider', { static: true }),
            __metadata("design:type", Object)
        ], SliderComponent.prototype, "sliderElement", void 0);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], SliderComponent.prototype, "style", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], SliderComponent.prototype, "value", null);
        __decorate([
            core.Output(),
            __metadata("design:type", Object)
        ], SliderComponent.prototype, "valueChange", void 0);
        __decorate([
            core.Input(),
            __metadata("design:type", Number),
            __metadata("design:paramtypes", [Number])
        ], SliderComponent.prototype, "min", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Number),
            __metadata("design:paramtypes", [Number])
        ], SliderComponent.prototype, "max", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Number),
            __metadata("design:paramtypes", [Number])
        ], SliderComponent.prototype, "step", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Number),
            __metadata("design:paramtypes", [Number])
        ], SliderComponent.prototype, "precision", null);
        __decorate([
            core.Input(),
            __metadata("design:type", String),
            __metadata("design:paramtypes", [String])
        ], SliderComponent.prototype, "orientation", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], SliderComponent.prototype, "range", null);
        __decorate([
            core.Input(),
            __metadata("design:type", String),
            __metadata("design:paramtypes", [String])
        ], SliderComponent.prototype, "selection", null);
        __decorate([
            core.Input(),
            __metadata("design:type", String),
            __metadata("design:paramtypes", [String])
        ], SliderComponent.prototype, "tooltip", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], SliderComponent.prototype, "tooltipSplit", null);
        __decorate([
            core.Input(),
            __metadata("design:type", String),
            __metadata("design:paramtypes", [String])
        ], SliderComponent.prototype, "tooltipPosition", null);
        __decorate([
            core.Input(),
            __metadata("design:type", String),
            __metadata("design:paramtypes", [String])
        ], SliderComponent.prototype, "handle", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], SliderComponent.prototype, "reversed", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], SliderComponent.prototype, "rtl", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], SliderComponent.prototype, "enabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], SliderComponent.prototype, "naturalArrowKeys", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Array),
            __metadata("design:paramtypes", [Array])
        ], SliderComponent.prototype, "ticks", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Array),
            __metadata("design:paramtypes", [Array])
        ], SliderComponent.prototype, "ticksPositions", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Array),
            __metadata("design:paramtypes", [Array])
        ], SliderComponent.prototype, "ticksLabels", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Number),
            __metadata("design:paramtypes", [Number])
        ], SliderComponent.prototype, "ticksSnapBounds", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], SliderComponent.prototype, "ticksTooltip", null);
        __decorate([
            core.Input(),
            __metadata("design:type", String),
            __metadata("design:paramtypes", [String])
        ], SliderComponent.prototype, "scale", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Array),
            __metadata("design:paramtypes", [Array])
        ], SliderComponent.prototype, "labelledBy", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Array),
            __metadata("design:paramtypes", [Array])
        ], SliderComponent.prototype, "rangeHighlights", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Function])
        ], SliderComponent.prototype, "formatter", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Function])
        ], SliderComponent.prototype, "lockToTicks", null);
        __decorate([
            core.Output(),
            __metadata("design:type", Object)
        ], SliderComponent.prototype, "slide", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", Object)
        ], SliderComponent.prototype, "slideStart", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", Object)
        ], SliderComponent.prototype, "slideStop", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", Object)
        ], SliderComponent.prototype, "change", void 0);
        SliderComponent = __decorate([
            core.Component({
                selector: 'mv-slider',
                template: "<input #slider>",
                styles: [""]
            }),
            __metadata("design:paramtypes", [])
        ], SliderComponent);
        return SliderComponent;
    }());

    var NgxBootstrapSliderModule = /** @class */ (function () {
        function NgxBootstrapSliderModule() {
        }
        NgxBootstrapSliderModule = __decorate([
            core.NgModule({
                imports: [],
                declarations: [NgxBootstrapSliderComponent, SliderComponent],
                exports: [
                    NgxBootstrapSliderComponent,
                    SliderComponent
                ]
            })
        ], NgxBootstrapSliderModule);
        return NgxBootstrapSliderModule;
    }());

    exports.NgxBootstrapSliderComponent = NgxBootstrapSliderComponent;
    exports.NgxBootstrapSliderModule = NgxBootstrapSliderModule;
    exports.NgxBootstrapSliderService = NgxBootstrapSliderService;
    exports.SliderComponent = SliderComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
